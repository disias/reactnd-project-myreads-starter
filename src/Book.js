import React,{Component} from 'react'
import PropTypes from 'prop-types'

class Book extends Component {

  static defaultProps = {
    element: null
  };

  static propTypes = {
    book: PropTypes.shape({
      title: PropTypes.string,
      authors: PropTypes.arrayOf(PropTypes.string),
      imageLinks: PropTypes.shape({thumbnail :PropTypes.string})}),
    element: PropTypes.element
  };

  render(){
    const book = this.props.book;
    const element = this.props.element;
    return (
        <div className="book">
        <div className="book-top">
          <div className="book-cover" style={{ width: 128, height: 192, backgroundImage: `url(${book.imageLinks !== undefined ? book.imageLinks.thumbnail: ''})`}} data-teste="imageLinks"></div>
            {element}
          </div>
        <div className="book-title" data-teste="title" >{book.title}</div>
        <div className="book-authors" data-teste="authors">{book.authors}</div>
      </div>
    )
  }
}



export class BookList extends Component {

  static defaultProps = {
    element: null
  };

  static propTypes = {
    books: PropTypes.arrayOf(
      PropTypes.shape({
      title: PropTypes.string,
      authors: PropTypes.arrayOf(PropTypes.string),
      imageLinks: PropTypes.shape({thumbnail :PropTypes.string})})),
    element: PropTypes.element
  };

  render(){
    const element = this.props.element;
    const books   =  this.props.books.map(book =>(
      <li key={book.id} ><Book book={book} element={element} /></li>
    ))

    return (
      <div className="bookshelf-books">
        <ol className="books-grid">
          {books}
        </ol>
      </div>
    )
  }
}

export default Book
