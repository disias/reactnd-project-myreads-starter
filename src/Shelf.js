import React ,{Component}  from 'react'
import {BookList} from './Book'
import {Link} from 'react-router-dom'

class Shelf extends Component {
  render(){
    let lastCategory = this.props.books[0].shelf;
    let books = [];
    return (
      <div className="list-books">
          <div className="list-books-title">
            <h1>{this.props.title}</h1>
          </div>
        <div className="list-books-content">
          <div>
              {this.props.books.map( (book,index,array) => {
                  let element = null
                  if(lastCategory !== book.shelf || index+1 === array.length  ){
                    if (index+1 === array.length){
                      books.push(book)
                    }
                    element = <BookShelf key={lastCategory} bookcategory={lastCategory} booklist={books}/>
                    lastCategory = book.shelf
                    books = []
                  }
                  if (index+1 !== array.length){
                    books.push(book)
                  }
                  return element
              })}
          </div>
        </div>

        <div className="open-search">
            <Link to='/search' >Add a book</Link>
        </div>
      </div>

    )
  }
}


export class BookShelf extends Component {
  render(){
    return (
      <div className="bookshelf">
        <BookShelfCategory title={this.props.bookcategory} />
        <BookList books={this.props.booklist}/>
     </div>
    )
  }
}


export class BookShelfCategory extends Component{
  render(){
    return (
      <h2 className="bookshelf-title">{this.props.title}</h2>
    )
  }
}


export class BookShelfCategoryList extends Component {
  render(){
    return (
      <div className="book-shelf-changer">
      <select>
        <option value="none" disabled>Move to...</option>
        <option value="currentlyReading">Currently Reading</option>
        <option value="wantToRead">Want to Read</option>
        <option value="read">Read</option>
        <option value="none">None</option>
      </select>
    </div>
    )
  }
}


export class ShelfSearchBook extends Component {
  state = {
    books: []
  }

  render(){
    return (
      <div className="search-books">
        <div className="search-books-bar">
          <Link to='/' className="close-search">Close</Link>
          <div className="search-books-input-wrapper">
            <input type="text" placeholder="Search by title or author"/>
          </div>
        </div>
        <div className="search-books-results">
            <BookList books={this.state.books}/>
        </div>
    </div>
    )
  }
}

export default Shelf
