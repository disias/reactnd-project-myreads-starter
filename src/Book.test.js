import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'
import {shallow} from 'enzyme'
import Book, { BookList } from './Book';
import {ShelfBookCategoryList}  from './Shelf'

/**
 This course is not designed to teach Test Driven Development.
 Feel free to use this file to test your application, but it
 is not required.
**/

const element = <ShelfBookCategoryList />


describe('Book', () => {
    const book = {
    id:1,
    title:'To Kill a Mockingbird',
    authors:['Harper Lee'],
    imageLinks: {
    thumbnail:'http://books.google.com/books/content?id=PGR2AwAAQBAJ&printsec=frontcover&img=1&zoom=1&imgtk=AFLRE73-GnPVEyb7MOCxDzOYF1PTQRuf6nCss9LMNOSWBpxBrz8Pm2_mFtWMMg_Y1dx92HT7cUoQBeSWjs3oEztBVhUeDFQX6-tWlWz1-feexS0mlJPjotcwFqAg6hBYDXuK_bkyHD-y&source=gbs_api'},
    shelf:'Currently Reading'};

    it('renders without crashing', () => {
      const div = document.createElement('div')
      ReactDOM.render(<Book book={book} />, div)
    })

    // it('testando se renderizou o title usando className', () => {
    //   const wrapper = shallow(<Book book={book} /> )
    //   expect(wrapper.find('.book-title').text()).toEqual(book.title);
    // })

    // it('testando se renderizou o title usando contains', () => {
    //    const wrapper = shallow(<Book book={book} /> )
    //    expect(wrapper.contains(book.title)).toBe(true)
    // })

    // it('testando se renderizou o authors usando className', () => {
    //   const wrapper = shallow(<Book book={book} /> )
    //   expect(wrapper.find('.book-authors').text()).toEqual(...book.authors);
    // })

    // it('testando se renderizou o authors usando contains', () => {
    //    const wrapper = shallow(<Book book={book} /> )
    //    expect(wrapper.contains(...book.authors)).toBe(true)
    // })

    it('testando se renderizou o title usando selector', () => {
      const wrapper = shallow(<Book book={book} /> )
      expect(wrapper.find('[data-teste="title"]').text()).toEqual(book.title);
    })

    it('testando se renderizou o authors usando selector', () => {
      const wrapper = shallow(<Book book={book} /> )
      expect(wrapper.find('[data-teste="authors"]').text()).toEqual(...book.authors);
    })

    it('testando se renderizou o imageLinks usando selector', () => {
      const wrapper = shallow(<Book book={book} /> )
      expect(wrapper.find('[data-teste="imageLinks"]').prop('style')).toHaveProperty('backgroundImage',`url(${book.imageLinks.thumbnail})`);
    })

    it('testando se renderizou o element', () => {
      const wrapper = shallow(<Book book={book} element={element} /> )
      expect(wrapper.contains(element)).toBe(true);
    })

})


describe('ListBook', () => {
  const books = [ {id:1,title:'To Kill a Mockingbird',authors:['Harper Lee'],imageLinks:{thumbnail:'http://books.google.com/books/content?id=PGR2AwAAQBAJ&printsec=frontcover&img=1&zoom=1&imgtk=AFLRE73-GnPVEyb7MOCxDzOYF1PTQRuf6nCss9LMNOSWBpxBrz8Pm2_mFtWMMg_Y1dx92HT7cUoQBeSWjs3oEztBVhUeDFQX6-tWlWz1-feexS0mlJPjotcwFqAg6hBYDXuK_bkyHD-y&source=gbs_api'},shelf:'Currently Reading'},
                  {id:2,title:'Enders Game',authors:['Orson Scott Card'],imageLinks:{thumbnail:'http://books.google.com/books/content?id=yDtCuFHXbAYC&printsec=frontcover&img=1&zoom=1&imgtk=AFLRE72RRiTR6U5OUg3IY_LpHTL2NztVWAuZYNFE8dUuC0VlYabeyegLzpAnDPeWxE6RHi0C2ehrR9Gv20LH2dtjpbcUcs8YnH5VCCAH0Y2ICaKOTvrZTCObQbsfp4UbDqQyGISCZfGN&source=gbs_api'},shelf:'Currently Reading'} ];

  it('renders booklist', () => {
      const output = renderer.create( <BookList books={books} element={element} />)
      expect(output).toMatchSnapshot()
  })

})
