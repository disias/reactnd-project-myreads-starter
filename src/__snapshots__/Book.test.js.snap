// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListBook renders booklist 1`] = `
<div
  className="bookshelf-books"
>
  <ol
    className="books-grid"
  >
    <li>
      <div
        className="book"
      >
        <div
          className="book-top"
        >
          <div
            className="book-cover"
            data-teste="imageLinks"
            style={
              Object {
                "backgroundImage": "url(http://books.google.com/books/content?id=PGR2AwAAQBAJ&printsec=frontcover&img=1&zoom=1&imgtk=AFLRE73-GnPVEyb7MOCxDzOYF1PTQRuf6nCss9LMNOSWBpxBrz8Pm2_mFtWMMg_Y1dx92HT7cUoQBeSWjs3oEztBVhUeDFQX6-tWlWz1-feexS0mlJPjotcwFqAg6hBYDXuK_bkyHD-y&source=gbs_api)",
                "height": 192,
                "width": 128,
              }
            }
          />
          <div
            className="book-shelf-changer"
          >
            <select>
              <option
                disabled={true}
                value="none"
              >
                Move to...
              </option>
              <option
                value="currentlyReading"
              >
                Currently Reading
              </option>
              <option
                value="wantToRead"
              >
                Want to Read
              </option>
              <option
                value="read"
              >
                Read
              </option>
              <option
                value="none"
              >
                None
              </option>
            </select>
          </div>
        </div>
        <div
          className="book-title"
          data-teste="title"
        >
          To Kill a Mockingbird
        </div>
        <div
          className="book-authors"
          data-teste="authors"
        >
          Harper Lee
        </div>
      </div>
    </li>
    <li>
      <div
        className="book"
      >
        <div
          className="book-top"
        >
          <div
            className="book-cover"
            data-teste="imageLinks"
            style={
              Object {
                "backgroundImage": "url(http://books.google.com/books/content?id=yDtCuFHXbAYC&printsec=frontcover&img=1&zoom=1&imgtk=AFLRE72RRiTR6U5OUg3IY_LpHTL2NztVWAuZYNFE8dUuC0VlYabeyegLzpAnDPeWxE6RHi0C2ehrR9Gv20LH2dtjpbcUcs8YnH5VCCAH0Y2ICaKOTvrZTCObQbsfp4UbDqQyGISCZfGN&source=gbs_api)",
                "height": 192,
                "width": 128,
              }
            }
          />
          <div
            className="book-shelf-changer"
          >
            <select>
              <option
                disabled={true}
                value="none"
              >
                Move to...
              </option>
              <option
                value="currentlyReading"
              >
                Currently Reading
              </option>
              <option
                value="wantToRead"
              >
                Want to Read
              </option>
              <option
                value="read"
              >
                Read
              </option>
              <option
                value="none"
              >
                None
              </option>
            </select>
          </div>
        </div>
        <div
          className="book-title"
          data-teste="title"
        >
          Enders Game
        </div>
        <div
          className="book-authors"
          data-teste="authors"
        >
          Orson Scott Card
        </div>
      </div>
    </li>
  </ol>
</div>
`;
